src/Nethermind/Nethermind.Blockchain/DevBlockProducer.cs

```diff
@@ -84,7 +85,7 @@ private Block PrepareBlock()
             header.TotalDifficulty = parent.TotalDifficulty + header.Difficulty;
             if (_logger.IsDebug) _logger.Debug($"Setting total difficulty to {parent.TotalDifficulty} + {header.Difficulty}.");
 
-            var transactions = _transactionStore.GetAllPending().OrderBy(t => t?.Nonce); // by nonce in case there are two transactions for the same account
+            var transactions = _transactionPool.PendingTransactions.OrderBy(t => t?.Nonce); // by nonce in case there are two transactions for the same account
```

Comment:
getting a list like property in a multithreaded context - makes it harder to clearly state who should synchronize - what do you think?

Reply:
I would leave it as a method that suggests that something happens underneath - some locking - so call it GetAllPending

Reply:
Ok, so I will change it to the be a method instead.